a3fe/configuration/engine_config.py

```diff
@@ -5,298 +5,243 @@
-    )
-    cutoff_distance: float = _Field(
-        10.0,  # Default to PME cutoff distance
-        alias="cutoff distance",
-        description="Cutoff distance in angstroms"
-    )
-    reaction_field_dielectric: float | None = _Field(
-        None,
-        alias="reaction field dielectric",
-        description="Reaction field dielectric constant(only for cutoffperiodic)"
-    )
-    @_validator('cutoff_type')
-    def _check_cutoff_type(cls, v):
-        if v not in ['PME', 'cutoffperiodic']:
-            raise ValueError('cutoff type must be PME or cutoffperiodic')
-        return v
-    
-    @_validator('cutoff_distance', always=True)
-    def _set_cutoff_distance(cls, v, values):
-        if values.get('cutoff_type') == 'PME':
-            return 10.0 if v is None else v
-        elif values.get('cutoff_type') == 'cutoffperiodic':
-            return 12.0 if v is None else v
-        return v
-    
-    @_validator('reaction_field_dielectric',always=True)
-    def _set_reaction_field_dielectric(cls, v, values):
-        cutoff_type = values.get('cutoff_type')
-        if cutoff_type == 'PME' and v is not None:
-            raise ValueError('reaction field dielectric should not be provided when cutoff type is PME')
-        elif cutoff_type == 'cutoffperiodic' and v is None:
-            return 78.3  # Default dielectric constant for cutoffperiodic
-        return v
+        # check temperature is in range [-200, 1000]
+        if not (-200 <= temperature <= 1000):
+            raise ValueError(f"Temperature must be between -200 and 1000 Celsius, got {temperature}")
 
-    @_field_validator("cutoff_distance", mode="before")
-    def _validate_cutoff_distance(cls, v, values):
-        """
-        Validate cutoff distance based on cutoff type.
-        """
-        cutoff_type = values.data.get("cutoff_type")
-        if cutoff_type == "cutoffperiodic":
-            return 12.0  # Default for cutoffperiodic
-        return v  # Default for PME (10.0)
+        # check pressure is in range [0, 1000] atm
+        if not (0 <= pressure <= 1000):
+            raise ValueError("pressure must be in range [0, 1000] atm.")
 
-    model_config = _ConfigDict(arbitrary_types_allowed=True)
-    
-    def __init__(self, stream_log_level: int = _logging.INFO, **data):
-        _BaseModel.__init__(self, **data)
-        _EngineRunnerConfig.__init__(self, stream_log_level=stream_log_level)
-        self._validate_runtime_and_update_config()
+        return v
 
-    def get_config(self) -> _Dict[str, _Any]:
+    @_model_validator(mode="after")
+    def _check_charge_and_cutoff(self):
         """
-        Get the SOMD configuration as a dictionary.
-
-        Returns
-        -------
-        config : Dict[str, Any]
-            The SOMD configuration dictionary.
+        Validate that if ligand_charge != 0, then cutoff_type must be PME.
         """
-        return self.model_dump()
+        ligand_charge = self.ligand_charge
```

Comment:
Probably just access on single line, rather than storing as variables (as you only use once)